### PR TITLE
haskell-stack 2.3.3

### PR DESCRIPTION
**Pull request by chenrui333**
Created with `brew bump-formula-pr`.

### CODE REVIEW

File: Formula/haskell-stack.rb [hunk 1]
```diff
@@ -5,8 +5,9 @@ class HaskellStack < Formula
 
   desc "The Haskell Tool Stack"
   homepage "https://haskellstack.org/"
-  url "https://github.com/commercialhaskell/stack/archive/v2.3.1.tar.gz"
-  sha256 "6701ddfc6d0be0c2bf0f75c84375e41923c5617f04222c5e582e7011c7f8fb83"
+  url "https://github.com/commercialhaskell/stack/releases/download/v2.3.3/stack-2.3.3-osx-x86_64.tar.gz"
```
Comment (dawidd6) [inline Formula/haskell-stack.rb:8]: This doesn't look like a correct url.

Comment (chenrui333) [reply Formula/haskell-stack.rb:8]: hmm, that is the one I actually used before. looks like the CI build failure might be because of this change.

**Comment by chenrui333** [comment]
remove ghc-8.6.x as not needed.

**Comment by chenrui333** [comment]
deprecate cabal-v1 install per #55253
And removed the bootstrapped artifacts

**Review by dawidd6 — APPROVED** [review]
Nice cleanup!

**Pull request closed without merging**